### DISCUSSION TITLE
New Issues Not Being Auto-Added to Project Board

### DISCUSSION THREAD

**Discussion by APrioriRainbows** [community/community#22477] [Projects and Issues]
<p>Hi,</p>
<p>I’m trying to use the built-in automation feature on Github projects boards on a Github Projects board.</p>
<p>I’ve created a number of issues, but they don’t seem to be showing up in the backlog column, which is set to the “to-do” automation. It was my understanding that they would be auto-added to that column when created. Am I missing something? This could be a permissions issue, but I don’t think so.</p>
<p>Any help is appreciated–Thank you!</p>
<p><img src="https://user-images.githubusercontent.com/14030441/181071315-12e2ce7f-a299-46d3-ba20-cf397c9cbd52.png" alt="Screenshot 2018-04-28 at 6.08.47 PM.png" data-base62-sha1="pmBAwivwm71NABkaWnGPriILUFz" width="453" height="368"></p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/apriorirainbows">@apriorirainbows</a>,</p>
<p>Just as a first check, when you are creating those new issues, are you also adding them to the Project? You can do so right in the issue, on the righthand side under “Projects”. You can find full instructions on adding an issue to a Project board from the side-bar <a href="https://help.github.com/articles/adding-issues-and-pull-requests-to-a-project-board/#adding-issues-and-pull-requests-to-a-project-board-from-the-sidebar" rel="nofollow noopener">here</a>.</p>
<p>If that doesn’t fix the issue for you, please do let me know and we’ll troubleshoot further.</p>

**Comment by APrioriRainbows** [comment]
<p>That was it! I get why that would not be the default since many people tend to have more projects so you wouldn’t know which project to add the issues too even if automation is on. But I guess I assumed since I had <strong>one</strong> project open and issues were automated to show up on it…they just would. Might be nice to add that as an option in cases like my own. Thanks!</p>

**Comment by nadiajoyce** [comment]
<p>So glad that solved it for you! And I completely understand what you mean. I’ll share that feedback with the appropriate teams in our internal feature request list.</p>

**Comment by ErikBjare** [comment]
<p>Could we have an option to automatically add an issue to a project when it’s created? I’d rather have all of them added by default and then remove them manually if I don’t want them there than the other way around.</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/erikbjare">@erikbjare</a>,</p>
<p>Thanks for this feedback! We’re always working to improve GitHub, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by jpike88** [comment]
<p>This would be greatly appreciated, manually shuffling new issues is a real pain and defeats the purpose of keeping everyone in projects instead of micro managing the issues list.</p>
<p>Additionally, is there a mailing list or notification we can recieve when this is implemented?</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/jpike88">@jpike88</a>,</p>
<p>Thanks for the additional thoughts! As mentioned in my other response here, we can’t guarantee anything. Though there isn’t a mailing list, I would recommend you follow <a href="https://blog.github.com/" rel="nofollow noopener">our blog</a>, as all new updates and releases will be announced there.</p>
<p>Cheers!</p>

**Comment by chaiwa-berian** [comment]
<p>Upvoting this request, and I can’t wait for this feature to be implemented! I had hard time moving one issue at a time to the project board.! Also, if they can make it so we are able to multi-select the issues and assign them to a given project, it would be much easier than the way it is currently implemented. The option to auto-assign issue to the repo’s project board would also be a great feature to have.</p>
<p>Peace! :slight_smile:</p>

**Comment by larsboehnke** [comment]
<p>I’d like to see especially those things:</p>
<ul>
<li>
<p>add issues/PRs automatically to a project</p>
</li>
<li>
<p>move issue automatically to ‘In Progress’ once a user is assigned</p>
</li>
<li>
<p>Have a QA step for functional review (when PR is reviewed, move issue to functional review instead of closing it or so, dunno)</p>
</li>
</ul>

**Comment by angela-crist** [comment]
<p>Hi <a class="mention" href="/larsboehnke">@larsboehnke</a>,</p>
<p>Thanks for the additional thoughts! Super helpful! As mentioned above we can’t promise any timeframes, however, the appropriate teams have been updated. </p>
<p>Cheers!</p>

**Comment by efcarey** [comment]
<p>It’s a bummer an issue can’t be auto added to a specific, or the only existing project, by default and the same problem with pull requests. But, more importantly, is there seriously no way to add/move more than 1 issue at a time to a project? </p>
<p>For example, say you use a project board to vizualize the status of issues in a current sprint (milestone). I can easily use filtering on the issues tab to see all the issues I want for a sprint, and even select all those issues using the single checkbox, and then assign ALL issues to a new milestone (sprint), this is good.</p>
<p>However, from what I can tell (I really hope I’m wrong) I either have to then manually open every single issue and assign it to a project, OR, use the “add card” function on the project board itself for EACH issue. I understand maybe the project board isn’t meant for visualizing sprints, but we really need the basic functionality of BULK moving issues to projects just like you can already with milestones, labels, assignments, etc. Am I wrong?</p>
<p>OR, really all a project needs is the SAME FILTERING OPTIONS available to you on the project page just like on the issues page. This way with a single click or two you can transform your project board into a global backlog, sprint backlog, sprint, etc and not have to move an issue INTO a project at all. This is where we are going to have to end up paying for a utlity like ZenHub which offers this exact functionality already.</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/efcarey">@efcarey</a></p>
<p>I totally understand why the ability to bulk add issues to a Project would be beneficial. This detailed feedback and ideas for solutions is really helpful, and I’m going to share it with the team in the existing issue for our internal feature request list. I still can’t make any promises, but I will be making sure the right people see this. Thanks!</p>

**Comment by efcarey** [comment]
<p>Thanks for replying <a class="mention" href="/nadiajoyce">@nadiajoyce</a>, I know github isn’t a PM tool but bulk moving of issues would solve literally all of my current road blocks. Bonus would be also to have the ‘project’ selection available in issue templates and PR templates so we can “auto-add to project” ourselves from issue creation by choosing the appropriate template.</p>
<p>Also, I thought for a second I could just use the filtering for the project board to handle my milestones, but it looks like milestones aren’t availble to be used as filters on the project board so that was a dead-end as well.</p>
<p>If there is a better place to submit this idea or are you aware of an issue already posted I can add my support to, please let me know, thanks!</p>

**Comment by nadiajoyce** [comment]
<p><a class="mention" href="/efcarey">@efcarey</a> Posting in the Community Forum with your ideas is a-okay! We do usually recommend people write into <a href="https://github.com/contact" rel="nofollow noopener">private support</a> with feature requests, but we do see lots of people posting here in the forum as well. We always take any feature reqeusts and feedback we see here and share it internally, so your ideas will get seen!</p>

**Comment by jholman** [comment]
<p>Just gonna add myself to the list of people who think it’s really important to be able to do something like this.</p>
<p>My situation: My team is dabbling with using GH for project management.  My developers all want to use Issues and PRs, because they want to be able to specify details, and they want to work at a per-repo level.  My boss likes Kanban boards, both because he likes the visual layout, and because he has more things he needs to juggle so he wants less detail.  The existence of organization-level Projects seemed like a two-birds-with-one-stone solution, because my boss conceptually groups multiple microservices as part of a single conceptual “chunk” of his architecture diagram (and he’s right).  And I don’t want to have to manually synchronize with Trello every day, and I’m too lazy to go look at Atlassian products.</p>
<p>Currently, every day I have to open every Issue and PR that might not have been added to the Project, and make sure it’s in the Project.  I can’t even see which ones need fixing without opening them individually!  This is currently a pain point that makes me wonder if I should look at some other tooling.</p>
<p>Potential solutions:</p>
<p>Ideally, I could automate this at a Repo level.  Ideally I could specify, per-repo, one or more Projects that Issues would be auto-added to, and the same thing for PRs.</p>
<p>Also ideally, in the list of Issues (ditto for PRs), there’d be a way to show what Projects an Issue has been assigned to.  Maybe a column where if there’s no Project it’s blank, and if there is a Project then you can hover to see a list, or something?  Heck, even just a number that counts how many Projects the Issue has been added to?</p>
<p>And then bulk Project-assignment (for both Issues and PRs) is another option, that could be an alternative to the auto-assignment feature I described above.  I imagine something like the way labels work in gmail?</p>

**Comment by deepakmahakale** [comment]
<p>You can provide similar feature as pull request templates. <br>
Allow us to add a file in git repository with the project name if it’s present you can auto-assign it </p>

**Comment by annie-l** [comment]
<p>would really love to have this too. or at least as <a class="mention" href="/jholman">@jholman</a> suggests, an ability to view, filter and/or bulk update from the issues list. my current workaround is to filter by no:project to see new issues that were added, but i do need to keep an eye out for new issues that happened to be added to a different project than the one i want everything in.</p>

**Comment by cvermilion** [comment]
<p>Chiming in to say no auto-adding issues/PRs is my biggest blocker to wanting to use GH Projects as our team’s primary PM tool. We’ve been using <a href="http://Waffle.io" rel="nofollow noopener">Waffle.io</a> but they’ve announced shutdown, so we’re looking at alternatives.</p>
<p>Echoing what <a class="mention" href="/deepakmahakale">@deepakmahakale</a> said, one relatively simple implementation would be to allow issue/PR templates to set the project – you can already set other metadata this way, so this seems like a natural thing to allow. Beyond implementing the auto-add-to-project feature everyone here is asking for, setting the project in a template seems like an obvious thing template users would want more generally.</p>
<p>Cheers,</p>
<p>Chris</p>

**Comment by dklotz** [comment]
<p>Any updates on auto-addding new issues/PRs to projects, is this actively being worked on? I was initially really confused that the option to “Move all newly added issues and pull request here” does not actually do what it says.</p>
<p>The naming “project” for a “board” is also a bit confusing, since a lot people refer to a “repository” on GH as a “GitHub project” (and that aids in misunderstand what is meant by “when an issue is added to a project”).</p>

**Reply by phifuh** [reply]
it´s 2024 and its not implemented, I think it is safe to assume its not getting worked on :d

**Reply by dpetratsch** [reply]
Yes its a shame, to omit such an essential function...
